### PR TITLE
Ensure that only `<` can be used in `max_session_ttl` predicates

### PR DESCRIPTION
**Pull request by vitorenesduarte**
As discussed in #27, we want `max_session_ttl` to only contain `<` inequalities, so this PR removes the support for `>`, `==` and `!=`.

Since we can only use `<`, all predicates evaluate to true. For this reason, the tests had to be simplified (and are not as useful now).

---

A follow up PR will implement the solver idea proposed in #27 so that the exported policy simply says `max_session_ttl < N` (even if the predicate has `and`/`or` expressions).

### CODE REVIEW

File: predicate/solver/ast.py [hunk 1]
```diff
@@ -459,36 +459,13 @@ def new(
             + nanoseconds * NANOSECOND
         )
 
-    def __eq__(self, val):
```
Comment (xacrimon) [inline predicate/solver/ast.py:462]: Is there a chance want to use durations in some other way later?
I'd probably keep these in error in the solver if they are invalid in the current context.

**Review by xacrimon — CHANGES_REQUESTED** [review]

File: predicate/solver/ast.py (hunk 1 above)
Comment (vitorenesduarte) [reply predicate/solver/ast.py]: Added them back in fb3c45e9a3cdd4307075cd463933cd73dccdd399, throwing an error when an invalid `Option` is created.

**Review by xacrimon — APPROVED** [review]

**Review by klizhentas — CHANGES_REQUESTED** [review]
I think we should solve this problem by introducing a new base class LtDuration, that will remove the operators from the Duration. Duration can add all other operators. This will have an advantage of not adding the operators that won't work in the first place, instead of

**Comment by xacrimon** [comment]
> I think we should solve this problem by introducing a new base class LtDuration, that will remove the operators from the Duration. Duration can add all other operators. This will have an advantage of not adding the operators that won't work in the first place, instead of

I do worry a bit that we'd have to get the user to explicitly use this class correctly. Seems like a prime place to make a mistake.

**Comment by vitorenesduarte** [comment]
> > I think we should solve this problem by introducing a new base class LtDuration, that will remove the operators from the Duration. Duration can add all other operators. This will have an advantage of not adding the operators that won't work in the first place, instead of
> 
> I do worry a bit that we'd have to get the user to explicitly use this class correctly. Seems like a prime place to make a mistake.

I was also concerned with this but the user never has to interact with `LtDuration`.
The user calls `Duration.new` which returns a `DurationLiteral`, not a `Duration` (and I believe this is where the confusion came from for me).

If we think this is confusing, we could e.g. move the `new` method out of `Duration` and call it `duration`, so the user would write
```python
Options.max_session_ttl < duration(hours=3h)
```
instead of
```python
Options.max_session_ttl < Duration.new(hours=3h)
```

---

Given the above, I've added `LtDuration` in c43c26d38dd69368014ad73b52ce32c7fb24cdfb.
I've also removed the possibility of writing inequalities containing e.g. two `Duration`s as these were not being used.
(Also because it wasn't clear to me if we should allow inequalities containing a `LtDuration` and a `Duration`.)

So in the end we only allow:
- `LtDuration < DurationLiteral`
- `Duration < DurationLiteral`
- `Duration > DurationLiteral`
- `Duration == DurationLiteral`
- `Duration != DurationLiteral`

File: predicate/solver/ast.py [hunk 2]
```diff
@@ -439,7 +442,26 @@ def walk(self, fn):
         fn(self)
 
     def __str__(self):
-        return "duration({})".format(self.name)
+        return "ltduration({})".format(self.name)
+
+    def __lt__(self, val):
```
Comment (klizhentas) [inline predicate/solver/ast.py:447]: You can make it a type check: `def __lt__(self, val: DurationLiteral)` so the parser will implement this check for you

**Comment by klizhentas** [comment]
Looks good, but please take a look at my last comment about using type check instead of a runtime?

We can make it a type check: def __lt__(self, val: DurationLiteral) so the parser will implement this check for you

File: predicate/solver/ast.py (hunk 2 above)
Comment (vitorenesduarte) [reply predicate/solver/ast.py]: Thanks. Done in e7eb4c99f8ea99bad53470bca249bafa79444d38.

**Review by klizhentas — APPROVED** [review]

**Pull request merged**